cmds/tui.c

```diff
@@ -2346,12 +2362,149 @@ static void tui_window_help(void)
 	delwin(win);
 }
 
+static void display_graph_field(WINDOW *win, int *selected_field, bool *graph_field_flag)
+{
+	int i;
+
+	for (i = 0; i < NUM_GRAPH_FIELD; i++) {
+		if (i == *selected_field)
+			wattron(win, A_REVERSE);
+		mvwprintw(win, i + ARRAY_SIZE(graph_field_help) + 4, 2, "[ %c ] %s", graph_field_flag[i] ? 'x' : ' ', fields[i]);
+		wattroff(win, A_REVERSE);
+	}
+}
+
+static void update_graph_output_fields(bool *graph_field_flag) {
+	struct display_field *field;
+	int i;
+	bool exist;
+
+	for (i = 0; i < NUM_GRAPH_FIELD; i++) {
+		exist = false;
+		list_for_each_entry(field, &graph_output_fields, list) {
+			if (field->id == i && !graph_field_flag[i]) {
+				exist = true;
+				del_field(field);
+				break;
+			}
+		}
+
+		if (!exist && graph_field_flag[i])
+			add_field(&graph_output_fields, graph_field_table[i]);
+	}
+}
+
+static void tui_window_graph_field(bool *graph_field_flag, int *selected_field)
+{
+	WINDOW *win;
+	int w = 64;
+	int h = ARRAY_SIZE(fields) + ARRAY_SIZE(graph_field_help) + 6;
+	unsigned i;
+
+	if (w > COLS)
+		w = COLS;
+	if (h > LINES)
+		h = LINES;
+
+	win = newwin(h, w, (LINES - h) / 2, (COLS - w) / 2);
+	keypad(win, true);
+	wrefresh(win);
+	box(win, 0, 0);
+
+	mvwprintw(win, 1, 2, "Customize fields in graph mode");
+
+	for (i = 0; i < ARRAY_SIZE(graph_field_help); i++)
+		mvwprintw(win, i + 3, 2, "%-*.*s", w-3, w-3, graph_field_help[i]);
+
+	display_graph_field(win, selected_field, graph_field_flag);
+
+
+	mvwprintw(win, h-1, w-1, "");
+	wrefresh(win);
+
+	bool done = false;
+	while (!done) {
+		int k = wgetch(win);
+
+		switch (k) {
+			case 'k':
+			case KEY_UP:
```

Comment:
We don't indent the case label.  Please put it in the sample column as the switch.